pkg/cmd/server/api/validation/oauth.go

```diff
@@ -152,23 +153,55 @@ func ValidateLDAPIdentityProvider(provider *api.LDAPPasswordIdentityProvider, id
 	return validationResults
 }
 
-func ValidateRequestHeaderIdentityProvider(provider *api.RequestHeaderIdentityProvider, identityProvider api.IdentityProvider) fielderrors.ValidationErrorList {
-	allErrs := fielderrors.ValidationErrorList{}
+func ValidateRequestHeaderIdentityProvider(provider *api.RequestHeaderIdentityProvider, identityProvider api.IdentityProvider) ValidationResults {
+	validationResults := ValidationResults{}
 
 	if len(provider.ClientCA) > 0 {
-		allErrs = append(allErrs, ValidateFile(provider.ClientCA, "provider.clientCA")...)
+		validationResults.AddErrors(ValidateFile(provider.ClientCA, "provider.clientCA")...)
 	}
 	if len(provider.Headers) == 0 {
-		allErrs = append(allErrs, fielderrors.NewFieldRequired("provider.headers"))
+		validationResults.AddErrors(fielderrors.NewFieldRequired("provider.headers"))
 	}
-	if identityProvider.UseAsChallenger {
-		allErrs = append(allErrs, fielderrors.NewFieldInvalid("challenge", identityProvider.UseAsChallenger, "request header providers cannot be used for challenges"))
+	if identityProvider.UseAsChallenger && len(provider.ChallengeURL) == 0 {
+		validationResults.AddErrors(fielderrors.NewFieldRequired("provider.challengeURL"))
```

Comment:
Didn't you recently add a way upstream to indicate details about being required?  If so, add a TODO to use the new API after the rebase.